docs/pages/api.md

```diff
@@ -446,9 +470,12 @@ The exact transformations are described below.
 |Jaccard|N/A|0|1.0|
 |Hamming|N/A|0|1.0|
 |Cosine[^note-angular-cosine]|`cosine similarity + 1`|0|2|
+|Dot[^note-dot-product]|`Dot similarity + 1`|0|2| 
```

Comment:
I think there might be something wrong with the table formatting: 
<img width="1598" alt="image" src="https://github.com/alexklibisz/elastiknn/assets/8015228/d44d0b53-cd09-4db0-b06f-8cdc7923d877">

Also, if we end up using it, we should describe the updated transformation here: `max(0, 1 + dot product)`